File: client/app/models/sortAnalytics.ts
import { ESortType } from "./eSortType";

export class SortAnalytics {
  public sortType: string;
  public itemCount: number;
  public executionTime: number;
  public bestTimeComplexity: string;
  public worstTimeComplexity: string;
  public averageTimeComplexity: string;

  public constructor(
    sortType?: string,
    itemCount?: number, executionTime?: number,
    bestTimeComplexity?: string, worstTimeComplexity?: string, averageTimeComplexity?: string
  ) {
    this.sortType = sortType ?? ESortType.UNDEFINED.toString();
    this.itemCount = itemCount ?? 0;
    this.executionTime = executionTime ?? 0;
    this.bestTimeComplexity = bestTimeComplexity ?? 'NA';
    this.worstTimeComplexity = worstTimeComplexity ?? 'NA';
    this.averageTimeComplexity = averageTimeComplexity ?? 'NA';
  }
}

File: server/SortKata/Domain/Models/SortAnalytics.cs
namespace SortKata.Domain.Models {
    public class SortAnalytics {
        public ESortType SortType { get; set; }
        public int ItemCount { get; set; }
        public TimeSpan ExecutionTime { get; set; }
        public string BestTimeComplexity { get; set; }
        public string WorstTimeComplexity { get; set; }
        public string AverageTimeComplexity { get; set; }

        public SortAnalytics() {
            this.SortType = ESortType.UNDEFINED;
            this.ItemCount = 0;
            this.ExecutionTime = TimeSpan.Zero;
            this.BestTimeComplexity = string.Empty;
            this.WorstTimeComplexity = string.Empty;
            this.AverageTimeComplexity = string.Empty;
        }
    }
}
